languages/ruby/reference/exercise-concepts/flatten-array.md

```diff
@@ -0,0 +1,29 @@
+# Concepts of flatten-array
+
+## General
+
+- method definition: the student must implement the `flatten` method
+- method arguments (positional): `flatten` takes one argument
+- return values: `flatten` should return an array
+- class methods: one approach is the define a class `FlattenArray` with a class method `flatten`
+- module methods: one approach is to implement `FlattenArray` as a module
+- arrays: the input is an array, and student's should understand the basics of arrays
+- array flatten: this method is the obvious/preferred way to flatten an array
+- array compact: this method is the obvious/preferred way to get rid of the `nil`'s
+- nil: the input array contains `nil` values, and the student should understand what `nil` is
+
+## Approach: `flatten` as a class method
```

Comment:
Just as a small hint: I think the things listed here are describing the Ruby implementation of Concepts, not the actual Concepts themselves. As an example, I think "add `flatten` using `def self.flatten`" is the Concept "methods (or class methods)". What do you feel?

Reply:
Yes, the example you gave describes how one would implement the concept of "class methods" in Ruby.
I'm not sure if I've understood what the listed approaches are meant to look like - could you point me to a reference document?

Reply:
There is now a reference video: [The Anatomy of a Concept Exercise](https://youtu.be/gkbBqd7hPrA)